libsolidity/ast/AST.cpp

```diff
@@ -576,6 +591,22 @@ FunctionTypePointer EventDefinition::functionType(bool _internal) const
 		return nullptr;
 }
 
+bool EventDefinition::CompareBySignature::operator()(
+	EventDefinition const* _lhs,
+	EventDefinition const* _rhs
+) const
+{
+	return eventSignature(_lhs) < eventSignature(_rhs);
+}
+
+string const EventDefinition::CompareBySignature::eventSignature(EventDefinition const* _event) const
+{
+	auto const& function = _event->functionType(true);
+	solAssert(function, "");
```

Comment:
WTF, the function doesn't even return a reference :)

```suggestion
	FunctionType const* functionType = _event->functionType(true);
	solAssert(functionType);
```

Reply:
Ah sorry, I just copy-pasted without paying attention from here: 
https://github.com/ethereum/solidity/blob/e44cdc728b48018aa0d1b415f8a1b3095e51206d/libsolidity/ast/AST.cpp#L223-L224

I will change that as well.

Reply:
Oh, so it was already wrong there. Can you fix that as well?

Reply:
Fixed.